### ISSUE TITLE
tweet on tag/commit?

### ISSUE THREAD

**Issue by kratsg** [scikit-hep/pyhf#1505]
Maybe we should have it so that we automatically tweet from `@pyhf_` on a new tag? https://github.com/ethomson/send-tweet-action

**Comment by Saransh-cpp** [comment]
Hi @kratsg, I have been experimenting a lot with the Twitter API and GitHub Actions lately:) and I think that this can be a good first issue for me, can I work on this issue if everyone wants to go forward with this? And if you are not planning to take this up?

**Comment by kratsg** [comment]
Fine with me, but need @lukasheinrich and @matthewfeickert to sign off.

**Comment by lukasheinrich** [comment]
I think we'd be happy to review a PR!

**Comment by matthewfeickert** [comment]
> Maybe we should have it so that we automatically tweet from `@pyhf_` on a new tag? https://github.com/ethomson/send-tweet-action

I usually like the tweet to be a bit more than "pyhf `vX.Y.Z` is out" and have it contain some very short info as well. Would it be easy to customize this somehow? If not, I'm not sure this is a huge help to have. 

@kratsg do you have examples of this working nicely for other projects?

@Saransh-cpp :wave: Thanks for your interest here. As we might have some back and forth on this if you'd like to contribute to the project sooner we've got a [few open Issues that we've tagged with `good first issue`](https://github.com/scikit-hep/pyhf/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) that we're happy to give more details on.

**Comment by kratsg** [comment]
> I usually like the tweet to be a bit more than "pyhf `vX.Y.Z` is out" and have it contain some very short info as well. Would it be easy to customize this somehow? If not, I'm not sure this is a huge help to have.

No good examples, but maybe we only have it trigger on releases and summarize the release information or just use the first sentence to fit into a tweet. Only issue is that since tweets are limited in # characters, we'd have to make sure that the first sentence of a release summarizes nicely what's in it in the right amount of characters -- or not... Not so easy to customize unless we do something like this?

**Comment by matthewfeickert** [comment]
> Only issue is that since tweets are limited in # characters, we'd have to make sure that the first sentence of a release summarizes nicely what's in it in the right amount of characters -- or not... Not so easy to customize unless we do something like this?

Yeah, I'm not against having automated tweets, but for a version release I don't really know if this gets us anything? Out of the things that are on [the release checklist](https://github.com/scikit-hep/pyhf/blob/106263debda987ecee403aeeab06726de4e8865e/.github/ISSUE_TEMPLATE/~release-checklist.md) writing a Tweet is pretty easy. Also, we really don't want to tweet that there is a new release out until it is out everywhere across PyPI and Conda-forge and the Conda-forge release can take multiple hours after the PyPI release is out. :/

**Comment by Saransh-cpp** [comment]
> @Saransh-cpp 👋 Thanks for your interest here. As we might have some back and forth on this if you'd like to contribute to the project sooner we've got a [few open Issues that we've tagged with `good first issue`](https://github.com/scikit-hep/pyhf/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22) that we're happy to give more details on.

Yes, I see, this might require some more discussion, I'll look for another issue to proceed with. Thanks for your time, everyone.
